contracts/MetaverseBaseNFT.sol

```diff
@@ -231,22 +228,26 @@ contract MetaverseBaseNFT is
     // function to disable gasless listings for security in case
     // opensea ever shuts down or is compromised
     // from CryptoCoven https://etherscan.io/address/0x5180db8f5c931aae63c74266b211f580155ecac8#code
-    function setIsOpenSeaProxyActive(bool _isOpenSeaProxyActive) public onlyOwner {
+    function setIsOpenSeaProxyActive(bool _isOpenSeaProxyActive)
+        public
+        onlyOwner
+    {
         isOpenSeaProxyActive = _isOpenSeaProxyActive;
     }
 
     // ---- Minting ----
 
-    function _mintConsecutive(uint256 nTokens, address to, bytes32 extraData) internal {
-        require(_tokenIdCounter.current() + nTokens + reserved <= maxSupply, "Not enough Tokens left.");
+    function _mintConsecutive(
+        uint256 nTokens,
+        address to,
+        bytes memory extraData
+    ) internal {
+        require(
+            _totalMinted() + nTokens + reserved <= maxSupply,
+            "Not enough Tokens left."
+        );
 
-        for (uint256 i; i < nTokens; i++) {
-            uint256 tokenId = _tokenIdCounter.current();
-            _tokenIdCounter.increment();
-
-            _safeMint(to, tokenId);
-            data[tokenId] = extraData;
-        }
+        _safeMint(to, nTokens, extraData);
```

Comment:
extraData is some values ought to be stored in contract, available to be read later